torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -41,6 +42,22 @@
 
 log = logging.getLogger(__name__)
 
+GEMM_TEMPLATE_INIT_BLOCKING_SMALL_M = r"""
+    constexpr int64_t N = {{N}};
+    constexpr int64_t K = {{K}};
+    constexpr int64_t Mr = {{micro_gemm.register_blocking.block_m}};
+    constexpr int64_t Nr = {{micro_gemm.register_blocking.block_n}};
+    constexpr int64_t Kr = {{micro_gemm.register_blocking.block_k}};
+    constexpr int64_t Nr_blocks = (N + Nr - 1) / Nr;
+    constexpr int64_t Kr_blocks = (K + Kr - 1) / Kr;
+{%- if is_dynamic_M %}
+    const int64_t M = {{kernel.size(GemmOut, 0)}};
+{%- else %}
+    constexpr int64_t M = {{kernel.size(GemmOut, 0)}};
+{%- endif %}
+    const int64_t Mr_blocks = (M + Mr - 1) / Mr;
+"""
```

Comment:
Don't we need to consider thread blocking and cache blocking for SMALL_M ? Will this not achieve the best performance when K is large and N is not large enough ?

Reply:
The `micro-kernel`, `macro-kernel` and the outer loops (CPP template) of the GEMM in this PR are similar to those in Intel Extension for PyTorch.
I had added this code initially when it was unclear why Intel Extension for PyTorch was performing better than PyTorch. Since it turns out that the missing recipe is prefetching the next block of B, eventually, I'd reuse the existing macro & micro-kernel for it & add prefetching to it. However, in order to do that, I'd have to modify the existing CPP template (the outer loops) & all macro and micro-kernels as well (and not just `CppMicroGemmFP32Vec`).